modules/katamari/src/main/ts/ephox/katamari/api/Type.ts

```diff
@@ -26,6 +26,21 @@ export const isString: (value: any) => value is string =
 export const isObject: (value: any) => value is Object =
   isType('object');
 
+const getPrototypeOf = Object.getPrototypeOf;
+
+export const isPlainObject = (value: unknown): value is Object => {
+  if (!isObject(value)) {
+    return false;
+  }
+
+  const proto = getPrototypeOf(value);
+  if (proto === Object.prototype) {
+    return true;
+  } else {
+    return proto.toString() === '[object Object]';
```

Comment:
How often is the toString branch triggered? Slate uses `isPlainObject` quite a lot and it becomes quite a performance bottleneck if used too frequently.